main.py

```diff
@@ -119,6 +121,27 @@ def move_log_file_to_experiment_folder(logger_id, experiment_path, filename, lev
     logger.add(new_log_path, level=level)
 
 
+def validate_scenario_list(scenario_params_list, experiment_path):
```

Comment:
Couldn't we save the `Scenarios` instantiated, with their data split, so that we don't re-instantiate them and re-split data?

Reply:
We could, but I think it would be a bit messy : some scenerio stuff would be done in the `validate_scenario_list` method (like `split_data()` for instance and other (like `preprocess_scenarios_data`) would be done in the `run_scenario `method

Reply:
What is the most Pythonic:

- do the scenario instanciation and the split data twice
- or do it once, and then do following operations outside of the validation loop

?

My intuition would be the second one. Maybe by just renaming your method to something like `instantiate_and_validate...` that would look clearer. No?

Reply:
Hey, I'm not sure about the most efficient way to organize it, maybe we could use `timeit`?

Reply:
We should not save the scenarios instanciated during the validation because, we would anyway need to re-create them for the other random states.

Reply:
And there is no need to explicitely call `del current_scenario`, as all the variables will be freed when the function will exit.

Reply:
Ok then